property_tests/arbitraries/candid/constructed/record_arb/base.ts

```diff
@@ -0,0 +1,110 @@
+import fc from 'fast-check';
+
+import { CandidMeta } from '../../candid_arb';
+import { CandidType } from '../../candid_type_arb';
+import { UniqueIdentifierArb } from '../../../unique_identifier_arb';
+import { JsFunctionNameArb } from '../../../js_function_name_arb';
+import { Record } from './index';
+
+type Field = [string, CandidMeta<CandidType>];
+
+export function RecordArb(candidTypeArb: fc.Arbitrary<CandidMeta<CandidType>>) {
+    return fc
+        .tuple(
+            UniqueIdentifierArb('typeDeclaration'),
+            fc.uniqueArray(fc.tuple(JsFunctionNameArb, candidTypeArb), {
```

Comment:
Is JsFunctionNameArb unique?

Reply:
Is that what uniqueArray is doing somehow? What happens if there are two JsFunctionNameArbs with the same name? Seems like we should be using UniqueIdentifierArb

Reply:
Unique Identifier Arb is going to make sure that globally all of the names are unique, for this case we don't need to them to be globally unique, we just need them to be locally unique to this particular record, another record could have fields of the same name and that would be okay. uniqueArray is what we want here.